File: client/src/data/entities/bingo-board-tile.ts
import { Tile } from './tile.ts';
import { TileSubmission } from './tile-submission.ts';

export interface BingoBoardTile {
  id: string,
  tile: Tile | null,
  position: GridPosition,
  submissions: TileSubmission[],
}

export interface GridPosition {
  column: number,
  row: number
}

File: API/Entities/BingoBoardTile.cs
namespace API.Entities;

public class BingoBoardTile
{
    public required string Id { get; set; }

    public required string BingoEventId { get; set; }
    public BingoEvent? BingoEvent { get; set; }  

    public required string TileId { get; set; }
    public Tile? Tile { get; set; }

    public required Position Position { get; set; }

    public IList<TileSubmission> Submissions { get; set; } = new List<TileSubmission>();
}

public class Position
{
    public required int Row { get; set; }
    public required int Column { get; set; }
}